modules/secrets-store-aws-secrets-manager.adoc

```diff
@@ -0,0 +1,332 @@
+----
+
+.. Create the provider resources by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f aws-provider.yaml
+----
+
+. Grant permission to allow the service account to read the AWS secret object:
+
+.. Create a directory to contain the credentials request by running the following command:
++
+[source,terminal]
+----
+$ mkdir credentialsrequest-dir-aws
+----
+
+.. Create a YAML file with the following configuration for the credentials request:
++
+.Example `credentialsrequest.yaml` file
+[source,yaml]
+----
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: aws-provider-test
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: AWSProviderSpec
+    statementEntries:
+    - action:
+      - "secretsmanager:GetSecretValue"
+      - "secretsmanager:DescribeSecret"
+      effect: Allow
+      resource: "arn:*:secretsmanager:*:*:secret:testSecret-??????"
+  secretRef:
+    name: aws-creds
+    namespace: test-namespace
+  serviceAccountNames:
+  - aws-provider
+----
++
+// TODO `resource` value: should we say something about this? I changed it from "secret:xxiaSecret" to "secret:testSecret". What are the "??????" supposed to be?
+
+.. Retrieve the OIDC provider by running the following command:
++
+[source,terminal]
+----
+$ oc get --raw=/.well-known/openid-configuration | jq -r '.issuer'
+----
++
+// TODO: Can this be rewritten to use jsonpath or something else instead of jq? (To avoid making users install jq)
++
+.Example output
+[source,terminal]
+----
+https://<name>-oidc.s3.<aws_region>.amazonaws.com
+----
+Copy the `<oidc_provider_arn>` from the output to use in the next step. For example, `<name>-oidc.s3.<aws_region>.amazonaws.com`.
+
+.. Use the `ccoctl` tool to process the credentials request by running the following command:
++
+[source,terminal]
+----
+$ ccoctl aws create-iam-roles \
+    --name test-role --region=<aws_region> \
+    --credentials-requests-dir=credentialsrequest-dir-aws \
+    --identity-provider-arn <oidc_provider_arn> --output-dir=credrequests-ccoctl-output
+----
++
+.Example output
+[source,terminal]
+----
+2023/05/15 18:10:34 Role arn:aws:iam::<aws_account_id>:role/test-role-test-namespace-aws-creds created
+2023/05/15 18:10:34 Saved credentials configuration to: credrequests-ccoctl-output/manifests/cert-manager-aws-creds-credentials.yaml
+2023/05/15 18:10:35 Updated Role policy for Role test-role-test-namespace-aws-creds
+----
++
+Copy the `<aws_role_arn>` from the output to use in the next step. For example, `arn:aws:iam::<aws_account_id>:role/test-role-test-namespace-aws-creds`.
+
+.. Bind the service account with the role ARN by running the following command:
++
+[source,terminal]
+----
+$ oc annotate sa/aws-provider eks.amazonaws.com/role-arn="<role_arn>"
+----
++
+// TODO: Do we need to pass in a particular ns?
```

Comment:
@xingxingxia I was comparing this with the cert-manager procedure that followed similar steps, and they had passed in a namespace with their command. Do we need to specify a namespace here too?

Reply:
The test case uses `oc new-project xxia-proj`, this command changes the context to the specific namespace, so all commands after this command are in the context of the specific namespace. So the test case does not explicitly set -n. You can set it:
```
oc annotate -n <test-namespace or my-namespace, unify it> sa/aws-provider eks.amazonaws.com/role-arn="<arn_role_arn> or <role_arn>, unify it"
```

Reply:
Thanks. So all of this should be done in the namespace then. I'll update to use either test-namespace or my-namespace throughout then